internal/rootcoord/meta_table.go

```diff
@@ -1036,13 +1037,74 @@ func (mt *MetaTable) unlockIsSegmentIndexed(segID typeutil.UniqueID, fieldSchema
 	return exist
 }
 
+func (mt *MetaTable) unlockGetCollectionInfo(collName string) (pb.CollectionInfo, error) {
+	collID, ok := mt.collName2ID[collName]
+	if !ok {
+		collID, ok = mt.collAlias2ID[collName]
+		if !ok {
+			return pb.CollectionInfo{}, fmt.Errorf("collection not found: %s", collName)
+		}
+	}
+	collMeta, ok := mt.collID2Meta[collID]
+	if !ok {
+		return pb.CollectionInfo{}, fmt.Errorf("collection not found: %s", collName)
+	}
+	return collMeta, nil
+}
+
+func (mt *MetaTable) checkFieldCanBeIndexed(collMeta pb.CollectionInfo, fieldSchema schemapb.FieldSchema, idxInfo *pb.IndexInfo) error {
+	for _, f := range collMeta.FieldIndexes {
+		if f.GetFiledID() == fieldSchema.GetFieldID() {
+			if info, ok := mt.indexID2Meta[f.GetIndexID()]; ok {
+				if idxInfo.GetIndexName() != info.GetIndexName() {
+					return fmt.Errorf(
+						"creating multiple indexes on same field is not supported, "+
+							"collection: %s, field: %s, index name: %s, new index name: %s",
+						collMeta.GetSchema().GetName(), fieldSchema.GetName(),
+						info.GetIndexName(), idxInfo.GetIndexName())
+				}
+			}
+			// TODO: unexpected: what if index id not exist? Meta incomplete.
+			msg := fmt.Sprintf("index meta was incomplete, index id missing in indexID2Meta, collection: %d, field: %d, index_id: %d",
+				collMeta.GetID(), fieldSchema.GetFieldID(), f.GetIndexID())
+			log.Warn(msg)
```

Comment:
If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.

Reply:
This information will be returned to the sdk, and the index id in the error info will not be easy to understand, so it is best to replace it with name

Reply:
> This information will be returned to the sdk, and the index id in the error info will not be easy to understand, so it is best to replace it with name

Agree.

Reply:
> If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.

`mt.indexID2Meta[f.GetIndexID()]` exist and `idxInfo.GetIndexName() == info.GetIndexName()` won't cause this function return error.

Reply:
> > If mt.indexID2Meta[f.GetIndexID()] exist, and idxInfo.GetIndexName() == info.GetIndexName(), error will also be returned. Func checkFieldIndexDuplicate wil not work.
> 
> `mt.indexID2Meta[f.GetIndexID()]` exist and `idxInfo.GetIndexName() == info.GetIndexName()` won't cause this function return error.

Sorry, my mistake, please ignore this.